File: src/main/java/com/example/chatService/repository/UserRepository.java
package com.example.chatService.repository;

import com.example.chatService.entity.User;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;

import java.util.Optional;

public interface UserRepository extends JpaRepository<User, String> {
    boolean existsByUsername(String username);
    boolean existsByEmail(String email);
    Optional<User> findByUsername(String username);
    Optional<User> findByEmail(String email); // 이메일로 조회
    Optional<User> findById(String id);

    @Query("select u.username from User u where u.id = :userId")
    String findUsernameById(String userId);
}

File: src/main/java/com/example/chatService/repository/UserRepository.kt
package com.example.chatService.repository

import com.example.chatService.entity.User
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query
import java.util.Optional

interface UserRepository : JpaRepository<User, String> {
    fun existsByUsernameValue(username: String): Boolean

    fun existsByEmail(email: String): Boolean

    fun findByEmail(email: String): Optional<User>

    @Query("select u.usernameValue from User u where u.id = :userId")
    fun findUsernameValueById(userId: String): String?
}
